src/main/java/com/google/codeu/servlets/StatsPageServlet.java

```diff
@@ -38,4 +40,62 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     jsonObject.addProperty("messageCount", messageCount);
     response.getOutputStream().println(jsonObject.toString());
   }
-}
\ No newline at end of file
+  
+  /*
+   * Obtaining the average message by going through all the list
+   */
+  public void avgMessage(HttpServletRequest request, HttpServletResponse response) 
```

Comment:
These aren't going to get called by JavaScript. doGet() is what is called. You can change this functions signature to return an int and return the average. Then you can call it from doGet() and add a property to the jsonObject in that method. You can do the same thing for the biggestMessage() method.